crates/iceberg/src/expr/predicate.rs

```diff
@@ -55,6 +60,29 @@ impl<T, const N: usize> LogicalExpression<T, N> {
     }
 }
 
+impl<T: Bind, const N: usize> Bind for LogicalExpression<T, N> {
+    type Bound = LogicalExpression<T::Bound, N>;
+
+    fn bind(self, schema: SchemaRef, case_sensitive: bool) -> Result<Self::Bound> {
+        let mut bound_inputs = MaybeUninit::<[Box<T::Bound>; N]>::uninit();
+        for (i, input) in self.inputs.into_iter().enumerate() {
+            let input = input.bind(schema.clone(), case_sensitive)?;
+            // SAFETY: The pointer is valid from [`MaybeUninit`].
+            unsafe {
```

Comment:
It feels like a bit of a premature optimisation to introduce `unsafe` and `MaybeUninit` here when safe code will probably do.